frontend/src/metabase/hoc/Favicon.jsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import _ from "underscore";
+import MetabaseSettings from "../lib/settings";
+
+const DEFAULT_FAVICON = MetabaseSettings.get("application-favicon-url");
+export const LOAD_COMPLETE_FAVICON = "app/assets/img/blue_check.png";
+
+let currentFavicon = DEFAULT_FAVICON;
+
+const updateFavicon = _.debounce(() => {
+  document
+    .querySelector('link[rel="icon"]')
+    .setAttribute("href", currentFavicon);
+});
+
+const favicon = faviconSetterOrGetter => ComposedComponent =>
```

Comment:
Could this be written using a functional component instead? E.g.

```
const withFavicon = selector => ComposedComponent => props => {
  const favicon = selector(props);

  useEffect(() => {
    document
      .querySelector('link[rel="icon"]')
      .setAttribute("href", favicon);
  }, [favicon]);

  return <WrappedComponent  {...props}    />;
};

export default withFavicon;
```

It would be also great to use TypeScript for a new component

Reply:
If there are issues with this change, could use use `componentDidMount` instead of `UNSAFE_` handler?